### ISSUE TITLE
IDE exception on opening of git context menu

### ISSUE THREAD

**Issue by Alexsey** [intellij-solidity/intellij-solidity#161]
1. Open IDE
2. Press `Menu` button (or right click)
3. Press `G` (or move the mouse on git)

EB: Everything should be OK
CB: IDE Exceptions indicator (on the bottom right) starts to gleam red

Plugin version: 2.1.3
IDE version: WebStorm 2018.3 EAP
Build #WS-183.4284.46, built on November 8, 2018
WebStorm EAP User
Expiration date: December 8, 2018
JRE: 1.8.0_152-release-1343-b15 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

Stacktrace:
```
com.intellij.openapi.extensions.impl.PicoPluginExtensionInitializationException: 0
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentInstance(ExtensionComponentAdapter.java:90)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getExtension(ExtensionComponentAdapter.java:113)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:252)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensionList(ExtensionPointImpl.java:179)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensionList(ExtensionPointName.java:32)
	at com.intellij.execution.actions.RunConfigurationProducer.getProducers(RunConfigurationProducer.java:34)
	at com.intellij.execution.actions.ConfigurationContext.findExisting(ConfigurationContext.java:237)
	at com.intellij.execution.actions.BaseRunConfigurationAction.update(BaseRunConfigurationAction.java:190)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:171)
	at com.intellij.openapi.actionSystem.impl.Utils.doUpdate(Utils.java:202)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:151)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:295)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.show(ActionPopupMenuImpl.java:109)
	at com.intellij.openapi.editor.impl.EditorImpl.invokePopupIfNeeded(EditorImpl.java:4633)
	at com.intellij.openapi.editor.impl.EditorImpl.access$9400(EditorImpl.java:124)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.runMousePressedCommand(EditorImpl.java:3700)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.mousePressed(EditorImpl.java:3624)
	at java.awt.Component.processMouseEvent(Component.java:6545)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6313)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4903)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at com.intellij.ide.actions.ShowPopupMenuAction.actionPerformed(ShowPopupMenuAction.java:47)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:258)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:275)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:620)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$processAction$2(IdeKeyEventDispatcher.java:669)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:668)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.java:520)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:475)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:212)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:643)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at com.intellij.execution.actions.RunConfigurationProducer.<init>(RunConfigurationProducer.java:55)
	at me.serce.solidity.ide.run.SolidityRunConfigProducer.<init>(SolidityRunConfigProducer.kt:18)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:85)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:62)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:45)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentInstance(ExtensionComponentAdapter.java:72)
	... 46 more
Caused by:

java.lang.ArrayIndexOutOfBoundsException: 0
	at com.intellij.execution.actions.RunConfigurationProducer.<init>(RunConfigurationProducer.java:55)
	at me.serce.solidity.ide.run.SolidityRunConfigProducer.<init>(SolidityRunConfigProducer.kt:18)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:85)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:62)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:45)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentInstance(ExtensionComponentAdapter.java:72)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getExtension(ExtensionComponentAdapter.java:113)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:252)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensionList(ExtensionPointImpl.java:179)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensionList(ExtensionPointName.java:32)
	at com.intellij.execution.actions.RunConfigurationProducer.getProducers(RunConfigurationProducer.java:34)
	at com.intellij.execution.actions.ConfigurationContext.findExisting(ConfigurationContext.java:237)
	at com.intellij.execution.actions.BaseRunConfigurationAction.update(BaseRunConfigurationAction.java:190)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:171)
	at com.intellij.openapi.actionSystem.impl.Utils.doUpdate(Utils.java:202)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:151)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:177)
	at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:295)
	at com.intellij.openapi.actionSystem.impl.ActionPopupMenuImpl$MyMenu.show(ActionPopupMenuImpl.java:109)
	at com.intellij.openapi.editor.impl.EditorImpl.invokePopupIfNeeded(EditorImpl.java:4633)
	at com.intellij.openapi.editor.impl.EditorImpl.access$9400(EditorImpl.java:124)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.runMousePressedCommand(EditorImpl.java:3700)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.mousePressed(EditorImpl.java:3624)
	at java.awt.Component.processMouseEvent(Component.java:6545)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6313)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4903)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at com.intellij.ide.actions.ShowPopupMenuAction.actionPerformed(ShowPopupMenuAction.java:47)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:258)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:275)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:620)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$processAction$2(IdeKeyEventDispatcher.java:669)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:668)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.java:520)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:475)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:212)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:643)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

EDIT: update stacktrace. Was from wrong exception

**Comment by mbolotov** [comment]
@Alexsey 
I was unable to reproduce your issue locally, but I believe #163 will fix it

**Comment by Alexsey** [comment]
Ok, I'll try after the merge and write about the results

Actually, you don't even need to open the Git submenu - the exception occurs right after the opening of the context menu

**Comment by SerCeMan** [comment]
Hey, @Alexsey! I believe @mbolotov crafted a fix this issue in the new [`2.1.4` release](https://plugins.jetbrains.com/plugin/9475-intellij-solidity) 🎉! Please reopen the issue if you can still reproduce the error!

**Comment by Alexsey** [comment]
I can confirm that the bug has been fixed. Thank you!
